### ISSUE TITLE
Where and how to set the token?

### ISSUE THREAD

**Issue by Vad1mo** [openshift/jenkins-sync-plugin#118]
We are running an external jenkins that is not on openshift.  Never the less we would like to use the jenkins-sync-plugin. 

After installing and configuring the plugin we get this error. So the question is where and how do we set the token?

```
jenkins                | Feb 22, 2017 11:23:23 AM io.fabric8.jenkins.openshiftsync.BuildWatcher$1 doRun
jenkins                | SEVERE: Failed to load initial Builds: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://openshift.acme.com:8443/oapi/v1/namespaces/cd/builds?fieldSelector=status%3DNew. Message: User "system:anonymous" cannot list builds in project "cd". Received status: Status(apiVersion=v1, code=403, details=StatusDetails(causes=[], group=null, kind=builds, name=null, retryAfterSeconds=null, additionalProperties={}), kind=Status, message=User "system:anonymous" cannot list builds in project "cd", metadata=ListMeta(resourceVersion=null, selfLink=null, additionalProperties={}), reason=Forbidden, status=Failure, additionalProperties={}).
jenkins                | io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://openshift.acme.com:8443/oapi/v1/namespaces/cd/builds?fieldSelector=status%3DNew. Message: User "system:anonymous" cannot list builds in project "cd". Received status: Status(apiVersion=v1, code=403, details=StatusDetails(causes=[], group=null, kind=builds, name=null, retryAfterSeconds=null, additionalProperties={}), kind=Status, message=User "system:anonymous" cannot list builds in project "cd", metadata=ListMeta(resourceVersion=null, selfLink=null, additionalProperties={}), reason=Forbidden, status=Failure, additionalProperties={}).
jenkins                |        at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:315)
jenkins                |        at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:268)
jenkins                |        at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:237)
jenkins                |        at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:230)
jenkins                |        at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:481)
jenkins                |        at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:60)
jenkins                |        at io.fabric8.jenkins.openshiftsync.BuildWatcher$1.doRun(BuildWatcher.java:78)
jenkins                |        at hudson.triggers.SafeTimerTask.run(SafeTimerTask.java:50)
jenkins                |        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
jenkins                |        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
jenkins                |        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
jenkins                |        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
jenkins                |        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
jenkins                |        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
jenkins                |        at java.lang.Thread.run(Thread.java:745)

```

**Comment by jimmidyson** [comment]
Internally this uses the fabric8 kubernetes client so you can set it via system property or environment variable - see https://github.com/fabric8io/kubernetes-client/blob/master/README.md#configuring-the-client for more details.

**Comment by gabemontero** [comment]
hey @Vad1mo - Starting with commit https://github.com/openshift/jenkins-sync-plugin/commit/205061c0aaa3ac4bb67ed4d1728d160aaa59e73d we now support the sync plugin leveraging Jenkins credentials.  You can now define a credential to include your token and then configure sync plugin to use that token.

The feature has been available since v0.1.13, though as of this writing we are up to v0.1.16, and we should be cutting the next version in a matter of days.
